pkg/kademlia/dialer.go

```diff
@@ -109,6 +109,23 @@ func (dialer *Dialer) FetchPeerIdentity(ctx context.Context, target pb.Node) (pI
 	return identity.PeerIdentityFromPeer(p)
 }
 
+// Info connects to a node address and returns its node info.
+func (dialer *Dialer) Info(ctx context.Context, address *pb.NodeAddress) (*pb.InfoResponse, error) {
```

Comment:
Just realizing that having `Dialer` with methods `FetchPeerIdentity` and `Info` seems weird, seems like it's being done this way because it's convenient rather than clear separation of concerns. `Dialer` was meant mainly for handling kademlia communication protocol, nothing else. We can think how to handle that later.

Also instead of `Info`, maybe use method name `FetchInfo` to make clearer that it's making a request.

Reply:
Changed in latest commit.